### PR TITLE
authorization: clean up api

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: pkg/net/grpc/auth.go [hunk 1]
```diff
@@ -156,3 +142,20 @@ func OwnerFromTokenMD(ctx context.Context, ownerClaim string) (string, error) {
 	}
 	return owner, err
 }
+
+// OwnerFromTokenMD is a helper function for extracting the sub claim from the :authorization gRPC metadata of the request.
```
Comment (Danielius1922) [inline pkg/net/grpc/auth.go]: SubjectFromTokenMD

File: authorization/service/deleteDevices_test.go [hunk 2]
```diff
@@ -6,10 +6,14 @@ import (
 	"testing"
 
 	"github.com/plgd-dev/cloud/authorization/pb"
+	"github.com/plgd-dev/cloud/pkg/net/grpc"
 	kitNetGrpc "github.com/plgd-dev/cloud/pkg/net/grpc"
 	"github.com/stretchr/testify/require"
 )
 
+const jwtWithSubTestUser = `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0ZXN0VXNlcklEIn0.6EZJidMCJ5UMwyttpwUNer-GdsBmPH1_ckH8ZU-SRpo`
```
Comment (Danielius1922) [inline authorization/service/deleteDevices_test.go]: This works, though it's not obvious what does the token contain, so the programmer has to do an additional step and decode it if he wants to see. I'm thinking we could create a encoding function that takes map of claims and returns the encoded token as string - see https://pkg.go.dev/github.com/golang-jwt/jwt#example-New-Hmac

Comment (jkralik) [reply authorization/service/deleteDevices_test.go]: Thx - fixed

File: pkg/net/grpc/auth.go (hunk 1 above)
Comment (jkralik) [reply pkg/net/grpc/auth.go]: Fixed

**Review by Danielius1922 — APPROVED** [review]

File: authorization/pb/auth.proto [hunk 3]
```diff
@@ -4,20 +4,17 @@ package ocf.cloud.auth.pb;
 
 option go_package = "github.com/plgd-dev/cloud/authorization/pb;pb";
 
-message GetUserDevicesRequest {
-    repeated string user_ids_filter = 1;
+message GetOwnerDevicesRequest {
```
Comment (ondrejtomcik) [inline authorization/pb/auth.proto]: Why not just GetDevicesRequest?

File: authorization/service/addDevice.go [hunk 4]
```diff
@@ -2,33 +2,29 @@ package service
...
 	"google.golang.org/grpc/status"
 )
 
-const serviceOwner = "*"
-
-func (s *Service) publishDevicesRegistered(ctx context.Context, owner, sub string, deviceID []string) error {
-	// TODO: verify that s.ownerClaim and sub are filled in JWToken
-	//	- use s.ownerClaim value from JWT for DevicesUnregistered.Owner
-	//	- use 'sub' from JWT for AuditContext
+func (s *Service) publishDevicesRegistered(ctx context.Context, owner, userID string, deviceID []string) error {
```
Comment (ondrejtomcik) [inline authorization/service/addDevice.go:20]: Shouldn't it be ownerID?

File: authorization/service/getUserDevices.go [hunk 5]
```diff
@@ -42,48 +41,25 @@ func sendUserDevices(request *pb.GetUserDevicesRequest, srv pb.AuthorizationServ
...
 			return log.LogAndReturnError(status.Errorf(status.Convert(err).Code(), "cannot get user devices: %v", err))
 		}
 	}
 	return nil
 }
 
-// GetUserDevices returns a list of user's devices if the access token is valid.
-func (s *Service) GetUserDevices(request *pb.GetUserDevicesRequest, srv pb.AuthorizationService_GetUserDevicesServer) error {
+// GetOwnerDevices returns a list of user's devices if the access token is valid.
+func (s *Service) GetOwnerDevices(request *pb.GetOwnerDevicesRequest, srv pb.AuthorizationService_GetOwnerDevicesServer) error {
```
Comment (ondrejtomcik) [inline authorization/service/getUserDevices.go]: You didn't rename the file. But let's discuss if it's GetOwnerDevices or GetDevices.

File: resource-aggregate/events/resourceLinksSnapshotTaken.go [hunk 6]
```diff
@@ -160,9 +160,9 @@ func (e *ResourceLinksSnapshotTaken) Handle(ctx context.Context, iter eventstore
 }
 
 func (e *ResourceLinksSnapshotTaken) HandleCommand(ctx context.Context, cmd aggregate.Command, newVersion uint64) ([]eventstore.Event, error) {
```
Comment (ondrejtomcik) [inline resource-aggregate/events/resourceLinksSnapshotTaken.go:162]: Please fix bug "[BUG] AuditContect is null in ResourceLinksPublished events"

Comment (jkralik) [reply resource-aggregate/events/resourceLinksSnapshotTaken.go:162]: will be done in separate PR

File: authorization/service/getUserDevices.go (hunk 5 above)
Comment (Danielius1922) [reply authorization/service/getUserDevices.go]: `Owner` prefix removed types and functions; file renamed.

**Pull request merged**